airflow/contrib/operators/dataproc_operator.py

```diff
@@ -884,6 +885,48 @@ class DataProcPySparkOperator(BaseOperator):
     template_fields = ['arguments', 'job_name', 'cluster_name']
     ui_color = '#0273d4'
 
+    @staticmethod
+    def _generate_temp_filename(filename):
+        dt = time.strftime('%Y%m%d%H%M%S')
+        return "{}_{}_{}".format(dt, str(uuid.uuid1())[:8], ntpath.basename(filename))
+
+    """
+    Upload a local file to a Google Cloud Storage bucket
+    """
+    def _upload_file_temp(self, local_file):
+        temp_filename = self._generate_temp_filename(local_file)
+        bucket = DataProcHook(gcp_conn_id=self.gcp_conn_id)._get_field('temp_bucket')
+
+        if not bucket:
+            raise AirflowException(
+                "If you want Airflow to upload the local file to a temporary bucket, set the 'temp_bucket' key in "
+                "the connection string")
+
+        self.log.info("Uploading {} to {}/{}", local_file, bucket, temp_filename)
```

Comment:
It would be good if you can prefix `gs://` to the path where the file would be uploaded.

```python
self.log.info("Uploading {} to gs://{}/{}", local_file, bucket, temp_filename)
```